scripts/go.pl

```diff
@@ -81,13 +81,19 @@ sub cmd_go
 		Irssi::settings_get_bool('go_match_case_sensitive'),
 		Irssi::settings_get_bool('go_match_anchored'));
 
+	my @matches = [];
 	foreach my $w (Irssi::windows) {
 		my $name = $w->get_active_name();
-		if ($name =~ $re) {
+		if ($name =~ /$re$/) {
```

Comment:
Better make this if(fc $name eq fc $chan) {

Reply:
That would be different behavior from the partial match then. I suppose I could add it to both. I don't think I've ever seen an IRC channel w/ caps anyway though.

Reply:
How is line 87 supposed to work according to you? To me it looks like it should go to an exactly matched channel, as Nei wrote, currently it matches from the end, ie /go tt would go to #mutt instead of #ttcars

Reply:
Sorry, I misunderstood the point of your original suggestion. I thought that you were just aiming for case insensitivity. I have pushed a new commit with this code amended as per your proposal.